### PR TITLE
Finalize Champion Modes and Implement Result Logic

### PR DESCRIPTION
**Pull request by Hebelag**
Fixes 0:-1 timer problem and implemented most of the champion mode logic.

### CODE REVIEW

File: app/src/main/java/com/scp/leagueofquiz/entrypoint/championquiz/ChampionQuizViewModel.java [hunk 1]
```diff
@@ -35,9 +35,10 @@
   private final MutableLiveData<Integer> score;
   private final MutableLiveData<Integer> failedAttempts;
   private final MutableLiveData<Instant> startTime;
-  private final MutableLiveData<Duration> timer;
+  private final MutableLiveData<Duration> timerNonTime;
```
Comment (mvettosi) [inline app/src/main/java/com/scp/leagueofquiz/entrypoint/championquiz/ChampionQuizViewModel.java:38]: Why did you rename this field like this?

File: app/src/main/java/com/scp/leagueofquiz/entrypoint/quizresult/QuizResultFragment.java [hunk 2]
```diff
@@ -8,28 +8,104 @@
...
+  //  return new QuizResultFragment();
+  //}
+
+  private QuizResultFragmentBinding binding;
+  private QuizResultViewModel viewModel;
+
+  @Override
+  public void onCreate(@Nullable Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    viewModel = new ViewModelProvider(this).get(QuizResultViewModel.class);
```
Comment (mvettosi) [inline app/src/main/java/com/scp/leagueofquiz/entrypoint/quizresult/QuizResultFragment.java:39]: Are you sure the viewModel can be initialised in the `onCreate()` method? Not saying that it can't, just asking if you made sure.
I noticed that Android Studio by default placed this initialisation in the `onActivityCreated()` method, maybe that's the right place to do it?

File: app/src/main/java/com/scp/leagueofquiz/entrypoint/quizresult/QuizResultViewModel.java [hunk 3]
```diff
@@ -1,7 +1,58 @@
...
+    private final MutableLiveData<Integer> score;
+    private final MutableLiveData<Integer> failedAttempts;
+    private final MutableLiveData<Duration> timer;
+
+    public QuizResultViewModel(@NonNull Application application){
+        super(application);
+        applicationContext = application.getApplicationContext();
+        score = new MutableLiveData<>(0);
+        failedAttempts = new MutableLiveData<>(0);
+        timer = new MutableLiveData<>(Duration.ZERO);
```
Comment (mvettosi) [inline app/src/main/java/com/scp/leagueofquiz/entrypoint/quizresult/QuizResultViewModel.java:34]: Maybe using live data for these 3 fields is a bit overkill for the QuizResult page, because we won't have anything that will be changing these values while the screen is shown.
I think it would be better to have them as raw types (not wrapped in MutableLiveData), and simply have the QuizResultFragment set the UI values in the `onViewCreated()`, without any observer.

What do you think?

**Review by mvettosi — CHANGES_REQUESTED** [review]
Added a few questions, let me know what you think of them :smile:

File: app/src/main/java/com/scp/leagueofquiz/entrypoint/championquiz/ChampionQuizViewModel.java (hunk 1 above)
Comment (Hebelag) [reply app/src/main/java/com/scp/leagueofquiz/entrypoint/championquiz/ChampionQuizViewModel.java:38]: yeah I forgot to change that after I pushed

File: app/src/main/java/com/scp/leagueofquiz/entrypoint/quizresult/QuizResultFragment.java (hunk 2 above)
Comment (Hebelag) [reply app/src/main/java/com/scp/leagueofquiz/entrypoint/quizresult/QuizResultFragment.java:39]: Might be, I think I just copied it from the ChampionQuizFragment.java

File: app/src/main/java/com/scp/leagueofquiz/entrypoint/quizresult/QuizResultViewModel.java (hunk 3 above)
Comment (Hebelag) [reply app/src/main/java/com/scp/leagueofquiz/entrypoint/quizresult/QuizResultViewModel.java:34]: Yeah true, it's one final value, so why change it.

**Comment by Hebelag** [comment]
I just commented everything.
